components/tidb_query_executors/src/fast_hash_aggr_executor.rs

```diff
@@ -252,24 +286,34 @@ impl<Src: BatchExecutor> AggregationExecutorImpl<Src> for FastHashAggregationImp
         mut input_physical_columns: LazyBatchColumnVec,
         input_logical_rows: &[usize],
     ) -> Result<()> {
+        self.free_trace(self.states_offset_each_logical_row.len() * size_of::<usize>());
         // 1. Calculate which group each src row belongs to.
         self.states_offset_each_logical_row.clear();
+
+        let rows_len = input_logical_rows.len();
+
+        if rows_len > 0 && self.n_bytes == 0 {
```

Comment:
question: why the alloc_trace is called only when n_bytes == 0

Reply:
It's a one time  allocation, so we only count it the first time.